File: backend/src/Tradion.Api/Models/InvoiceStatus.cs
namespace Tradion.Api.Models;

public static class InvoiceStatus
{
    public const string Draft = "Draft";
    public const string WaitingPayment = "WaitingPayment";
    public const string Paid = "Paid";

    public static bool IsPaid(string? status) =>
        string.Equals((status ?? "").Trim(), Paid, StringComparison.OrdinalIgnoreCase);
}

File: frontend/ike-web/src/app/core/status/invoice-status.ts
export const InvoiceStatus = {
  paid: 'paid'
} as const;

export function normalizeInvoiceStatus(status?: string | null): string {
  return (status || '').trim().toLowerCase();
}

export function isInvoicePaid(status?: string | null): boolean {
  return normalizeInvoiceStatus(status) === InvoiceStatus.paid;
}
